zipper/types/response.go

```diff
@@ -135,12 +135,34 @@ func mergeFetchResponsesWithEqualStepTimes(m1, m2 *protov3.FetchResponse, uuid s
 	return nil
 }
 
+func mergeFetchResponsesWithUnequalStepTimes(m1, m2 *protov3.FetchResponse, uuid string) error {
```

Comment:
Again, this can be fixed after this PR, but my opinion on merge logic is that we should prefer metric with less NaNs - because it might be possible that on a one server metric was created with 1s:7d but on other - with 60s:30d - and if we see a lot of NaN in the most precise one, but not in other - we should just drop the most precise one.

I'd also suggest to look into how it's done in Metrictank for consolidation: https://github.com/grafana/metrictank/blob/a673544664502e47ec25798b323c56f3daaccf3a/consolidation/consolidate.go